convert-hf-to-gguf.py

```diff
@@ -1186,6 +1186,19 @@ def write_tensors(self):
 @Model.register("StableLmForCausalLM", "StableLMEpochForCausalLM", "LlavaStableLMEpochForCausalLM")
 class StableLMModel(Model):
     model_arch = gguf.MODEL_ARCH.STABLELM
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.model_arch = (
+            gguf.MODEL_ARCH.STABLELM
+            if self.hparams["num_hidden_layers"] < 40
+            else gguf.MODEL_ARCH.STABLELM2
+        )
+        self.gguf_writer = gguf.GGUFWriter(
+            self.fname_out,
+            gguf.MODEL_ARCH_NAMES[self.model_arch],
+            endianess=self.endianess,
+            use_temp_file=False,
+        )
```

Comment:
I'm wondering if it would be possible to use the same model arch for the new StableLM 2 12B model, and use optional tensors to do so.

This would reduce future maintenance burden by not duplicating the graph-building function again.

Reply:
I took a stab at this in 0ec53cf